ReSwiftDemo/ViewControllers/LoginViewController.swift

```diff
@@ -0,0 +1,51 @@
+import UIKit
+import ReSwift
+
+class LoginViewController: UIViewController {
+
+    @IBOutlet weak var loginTextField: UITextField!
+    @IBOutlet weak var passwordTextField: UITextField!
+    @IBOutlet weak var activityIndicator: UIActivityIndicatorView!
+    @IBOutlet weak var loginButton: UIButton!
+    
+    var store: Store<AppState>!
+    var sessionUserInteractions: SessionUserInteractions!
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        self.store.subscribe(self)
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        self.store.unsubscribe(self)
+    }
+
+    @IBAction func loginClicked(_ sender: Any) {
+        self.sessionUserInteractions.signIn(
+            username: self.loginTextField.text ?? "",
+            password: self.passwordTextField.text ?? "")
+    }
+}
+
+// MARK: state management
+extension LoginViewController: StoreSubscriber {
+    
+    func newState(state: AppState) {
+        if state.sessionState.inProgress {
```

Comment:
to mozna zamienic na `?:` :)

Reply:
Funkcje zawsze coś zwracają. To, że typ nazywa się `Void` to tylko zmyła bo zwraca zawsze pusty tuplet/krotkę `()` :)